community/kernel/src/main/java/org/neo4j/kernel/impl/pagecache/ConfiguringPageCacheFactory.java

```diff
@@ -51,6 +55,16 @@ public ConfiguringPageCacheFactory(
         this.config = config;
         this.tracer = tracer;
         this.log = log;
+        configuredPageSize = 0;
+    }
+
+    /**
+     * Sets the page size, used by the {@link StandalonePageCacheFactory} for testing purposes.
+     * @param pageSize The new page size.
+     */
+    public void setPageSize( int pageSize )
+    {
+        configuredPageSize = pageSize;
     }
```

Comment:
I'd do this juggling in a `TestConfiguringPageCacheFactory` that extends and overrides the original `ConfiguringPageCacheFactory`. The `PageCacheRule` can then use that when configuring its page cache.